include/seqan3/range/views/minimiser.hpp

```diff
@@ -0,0 +1,555 @@
+
+    //!\brief Pre-increment.
+    window_iterator & operator++() noexcept
+    {
+        get_minimiser();
+        return *this;
+    }
+
+    //!\brief Post-increment.
+    window_iterator operator++(int) noexcept
+    {
+        window_iterator tmp{*this};
+        get_minimiser();
+        return tmp;
+    }
+
+    //!\brief Return the minimiser.
+    value_type operator*() const noexcept
+    {
+        return minimiser_value;
+    }
+
+    /*!\brief Forward this iterator.
+     * \attention This function is only avaible if `it_t` models std::random_access_iterator.
+     */
+    window_iterator & operator+=(difference_type const skip) noexcept
+    //!\cond
+        requires std::random_access_iterator<it_t>
+    //!\endcond
+    {
+        get_minimiser(skip);
+        return *this;
+    }
+
+    /*!\brief Forward copy of this iterator.
+     * \attention This function is only avaible if `it_t` models std::random_access_iterator.
+     */
+    window_iterator operator+(difference_type const skip) const noexcept
+    //!\cond
+        requires std::random_access_iterator<it_t>
+    //!\endcond
+    {
+        window_iterator tmp{*this};
+        return tmp += skip;
+    }
+
+    /*!\brief Non-member operator+ delegates to non-friend operator+.
+     * \attention This function is only avaible if `it_t` models std::random_access_iterator.
+     */
+    friend window_iterator operator+(difference_type const skip, window_iterator const & it) noexcept
+    //!\cond
+        requires std::random_access_iterator<it_t>
+    //!\endcond
+    {
+        return it + skip;
+    }
+
+    /*!\brief Move the iterator by a given offset and return the corresponding minimiser.
+     * \attention This function is only avaible if `it_t` models std::random_access_iterator.
+     */
+    value_type operator[](difference_type const n)
+    //!\cond
+        requires std::random_access_iterator<it_t>
+    //!\endcond
+    {
+        window_left += n;
+        get_minimiser();
+        return operator*();
+    }
+
+private:
+    //!brief Iterator to last element in range.
+    sentinel_t last_elem;
+
+    //!brief Keeps track if the minimiser value changed.
+    bool minimiser_changed{false};
+
+    //!\brief The minimiser value.
+    size_t minimiser_value{0};
+
+    //!\brief Iterator to the leftmost element of one window.
+    it_t window_left;
+
+    //!\brief Iterator to the rightmost element of one window.
+    it_t window_right;
+
+    //!\brief The number of elements in one window.
+    uint32_t num_w_elems;
+
+    //!\brief Stored values per window.
+    std::deque<uint64_t> window_values;
+
+    //!\brief Increments iterator by 1.
+    void get_minimiser()
+    {
+        if (window_values.size() == 0)
+            window_first();
+        else
+        {
+            // Call next_minimiser until minimiser value changed or end of the underlying range is reached.
+            while((!minimiser_changed) && (std::ranges::next(window_right) != last_elem))
```

Comment:
you do not need to store the variable `minimiser_changed` (and this makes the iterator more light weigth) if you make `next_minimizer` return a boolean. The boolean indicates whether a new minimizer has been found or the end has been reached (`true`)
```suggestion
            while(!next_minimiser())
```
Then inside `next_minimiser`
add after advancing window_left and window_right
```cpp
if (window_right == las_elem)
    return true;
```
This also saves the call to `std::ranges::next`